src/utils/queries/utils.mjs

```diff
@@ -0,0 +1,48 @@
+import { VALID_JAVASCRIPT_PROPERTY } from './constants.mjs';
+import createQueries from './index.mjs';
+
+/**
+ * @param {import('@types/mdast').List} list
+ * @returns {0 | 1 | 2} confidence
+ *
+ * 0: This is not a typed list
+ * 1: This is a loosely typed list
+ * 2: This is a strongly typed list
+ */
+export const isTypedList = list => {
+  if (!list || list.type !== 'list') {
+    return 0;
+  }
+
+  const firstNode = list.children?.[0]?.children?.[0]?.children[0];
```

Comment:
Do we need that many nested `?.` this feels quite ... deep. Also curious to know what children at this depth are supposed to be and if this is guaranteed for all typed lists? Is this also going through how many elements?

Because if this is going through every node on a document, that's super expnesive btw.